components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementService.java

```diff
@@ -257,6 +257,21 @@ public String exportSPApplicationFromAppID(String applicationId, boolean exportS
         return null;
     }
 
+    /**
+     * Export Service Provider application using application ID.
+     *
+     * @param applicationId ID of the SP.
+     * @param exportSecrets Specify whether to export the secrets or not.
+     * @param tenantDomain  Tenant domain.
+     * @return Service Provider.
+     * @throws IdentityApplicationManagementException Identity Application Management Exception.
+     */
+    public ServiceProvider exportSPFromAppID(String applicationId, boolean exportSecrets,
+                                               String tenantDomain) throws IdentityApplicationManagementException {
+
+        return null;
```

Comment:
what are we doing here?

Reply:
Here we return a null right? then why are we taking these parameters (applicationId, exportSecrets, tenantDomain)?

How you export the SP using application ID?

Reply:
The method is overridden in the ApplicationManagementServiceImpl.java file to return a service provider.